spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BrokerEvent.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.listener;
+
+import java.util.Map;
+
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.amqp.event.AmqpEvent;
+import org.springframework.util.Assert;
+
+/**
+ * @author Gary Russell
+ * @since 2.1
+ *
+ */
+@SuppressWarnings("serial")
+public class BrokerEvent extends AmqpEvent {
```

Comment:
Are you going to fine-grain to the possible event types?

I mean that from the target application perspective it really not so useful just to catch all the possible `BrokerEvent` in one place. Even if we could have separate listeners for different routing keys, we still will emit only `BrokerEvent`.

Reply:
Well, the event selection in the configured event keys can be very narrow `user.password.changed` or wildcarded `user.#`.

I was thinking that further selection can be done in the listener...

```java
@EventListener(condition = "event.eventType == 'queue.created'")
```

I really don't think we want to create a large number of java events to map the broker events - we'd have to keep them in sync each time they add a new one.